package/MDAnalysis/analysis/rms.py

```diff
@@ -381,11 +394,15 @@ def __init__(self, atomgroup, reference=None, select='all',
 
             .. Note:: Experimental feature. Only limited error checking
                       implemented.
-        weights : {"mass", ``None``} or array_like (optional)
-             choose weights. With ``"mass"`` uses masses as weights; with ``None``
-             weigh each atom equally. If a float array of the same length as
+        weights : {"mass", ``None``} or array_like or list of array_like (optional)
+             choose weights. With ``"mass"`` uses masses as weights for both `select`
+             and `groupselections`; with ``None`` weigh each atom equally for both 
+             `select` and `groupselections`. If a float array of the same length as
              `atomgroup` is provided, use each element of the `array_like` as a
-             weight for the corresponding atom in `atomgroup`.
+             weight for the corresponding atom in `select`, and assume ``None`` 
+             for `groupselections`. 
+             If weights is a list or array_like or 'mass' or None of length 
+             (1 + length(`groupselections`)), apply the weights correspondingly.
```

Comment:
As discussed in #2577, I think it would be better to split `weights` into two attributes, say `weights` and `groupweights`. You can keep the default for the former as `None`, but have the latter default to say `False` so that can be caught and set to an appropriate value based on `weights` as with the current behaviour.